src/wp-admin/edit-form-advanced.php

```diff
@@ -491,6 +491,54 @@
 <?php
 if ( isset( $post_new_file ) && current_user_can( $post_type_object->cap->create_posts ) ) {
 	echo ' <a href="' . esc_url( admin_url( $post_new_file ) ) . '" class="page-title-action">' . esc_html( $post_type_object->labels->add_new ) . '</a>';
+	
+	/**
+	 * Adds Previous and Next buttons alongside Add New button.
+	 * 
+	 * @param bool 	Whether to show Previous and Next buttons. Default true.
+	 * @param $post_type
+	 *
+	 * @since CP-1.x.x
+	 */
+	if ( apply_filters( 'admin-post-navigation', true, $post_type_object->name ) ) {
```

Comment:
The doc block says exactly what the parameter is that's being passed. `$post_type_object->name` is the post type.

I pondered on the filter name for a while. Similar filters already in core have a `wp-` prefix. Obviously that's not right, but I don't know whether we should have a 'cp-' prefix or not. I'm certainly open to modifying the filter name.

Reply:
Why would anyone want to add their own markup for this? That makes no sense at all. Either someone wants it on or they don't. The type of filter I've used is actually used quite a bit on pages in the admin.

Reply:
Perhaps because the code to get next and previous is generic? No parameters for taxonomy and same_term, which might be taken from the Post List if it was filtered. I thought that was in the original petition. Or if the list is sorted, that affects next and previous.

Reply:
The functions `get_previous_post()` and `get_next_post()` actually use the filters `get_previous_post_sort`, `get_next_post_sort`, `get_previous_post_where`, and `get_next_post_where`, which themselves expose all the parameters you mention. No need to reinvent the wheel.

Reply:
I see your point about the filtering though.

Reply:
Still, this generic previous and next doesn't fulfill what the petition asked for:
> This would be convenient if your editing posts in a particular category instead of having to click back on the “Posts” link in the admin sidebar, then having to filter the posts by category again.

Reply:
You interpret that comment in a different way from me. On your interpretation, if I filter my list of posts by category and then go to edit one of the posts, the "Next" and "Previous" buttons should now change to work only within the category. But that would be absurd.

Usually the reason someone filters by category in the list of posts is simply to make it easier to find a specific post. It does not mean that they want CP to hold on to that method of ordering the posts when they move on to a single post screen. Indeed, that would mean that I would have to undo manual filtering of the posts list just to ensure the ordering was not changed.

This is similar to the problem I see with using the function `get_adjacent_post()`. It runs the risk of the buttons getting out of sync with the default ordering. My implementation allows for the using of the filters I've mentioned to change the default ordering, which would then be respected by the "Next" and "Previous" buttons and keep everything in sync. But maybe I've missed something about `get_adjacent_post()`, so I'll take another look.

Reply:
I should also point out that if someone wants to manually filter a list of posts by (say) category and then run through that list in order, it can be done simply by keeping that filtered list visible in one tab while editing specific posts in another tab. Then it's easy to go back to the previous filtered list. This is also theoretically true of the full list of posts, of course, but that is likely to get too long to make it practical, which is why adding these buttons makes sense.

Reply:
I thought `get_previous_post` calls (eventually) `get_adjacent_post`. The question was not about which function to use, but what parameters to pass.
I _thought_ that the petition specifically wanted to stay within a category and that is the available parameter. (but you'd have to do that only when filtered in the post list)

Reply:
> (but you'd have to do that only when filtered in the post list)

But that's the problem. So, as you say, you'd have to pass the parameter using local storage. But that would force that parameter to be used even if the user didn't intend that. As I said above:

> Usually the reason someone filters by category in the list of posts is simply to make it easier to find a specific post. It does not mean that they want CP to hold on to that method of ordering the posts when they move on to a single post screen. Indeed, that would mean that I would have to undo manual filtering of the posts list just to ensure the ordering for the purposes of the "Next" and "Previous" buttons was not changed.

So I don't see how what you say can be achieved without creating a whole new problem.